### PR TITLE
Unit 4.1 create new user component

### PR DESCRIPTION
**Pull request by nathanjophillips**
## The Purpose
- To learn how to work with nested components and binding data with Polymer
- To generate a working create user component to use in further testing throughout the build of this project

## What It Does
1. This imports `create-user` into `intern-app` to be displayed as a create user button.  Once clicked a single editable user profile can be added into the existing `userProfiles` array with a unique string id property. 

2. The properties inside `create-user` are contained in a user object to then be dispatched as a new <b>Custom Event</b> that's bound to be used inside `intern app`.

3. Inside `intern-app` a constructor is called to then bind the `newUser` to the event. It also adds a `addEventListener` and a `removeEventListener` thrugh `connectedCallback()` and `disconnectedCallback()` functions to keep track of the changing properties for the `newUser`	

4. It's then pushed to the existing array of `userProfiles` in `newUser(event)` while also checking if the current id string property is unique with `pushToArray()`. If it's not, it generates a new one to be used through `genID()`

## This accomplishes the following requirements given for  [Unit 4.1](https://github.com/phillnab/nathan-phillips-ux-iop/blob/master/issues/unit-4/4.1-polymer-components.md)
- The following HTML pages are represented in the Polymer app:
    - [x] user-list.html
    - [x] user-profile.html
    - [x] create-user.html
    - [x] edit-user.html
- [x] The `user-list` element has `user-profile` components nested inside it
- [x] The `user-profile` component maintains its expand and collapse functionality

# How To Test
- Clone Repo [nathan-phillips-ux-iop](https://github.com/phillnab/nathan-phillips-ux-iop)
- Checkout branch `nathan-phillips-ux-iop/unit-4.1-create-user`
- Run `yarn`
- Run `yarn build`
- Run `yarn start`

### Step 1 
- See if it matches this:

Screen Shot 2019-05-23 at 1 51 09 PM

### Step 2 
- Click `Create User` and change default data and `Save`:

Screen Shot 2019-05-23 at 1 51 24 PM

Screen Shot 2019-05-23 at 1 51 58 PM

### Step 3 
- See if it Saved the input and added the new created user to the bottom of the userlist:

Screen Shot 2019-05-23 at 1 52 06 PM

## Checklist

Under penalty of public shaming, I avow that I:

- [x] Reviewed the diff to look for typos, whitespace errors, and console.log() statements
- [x] Verified that there are no compiler or linter errors or warnings
- [ ] Verified the build in production(optimized) mode
- [x] Updated the docs, if necessary
- [x] Included the appropriate labels
- [x] Referenced applicable issues (i.e. Closes #XXXX, Fixes #XXXX)

Browsers tested:
- [x] Chrome
- [ ] Firefox
- [ ] Safari
- [ ] Edge
- [ ] Internet Explorer

### CODE REVIEW

File: src/components/create-user.html [hunk 1]
```diff
@@ -44,31 +42,90 @@
...
-          <input type="text" name="phone" id="phone">
-        </div>
-        <div>
-          <label for="email">Email:</label>
-          <input type="text" name="email" id="email">
-        </div>
-        <button type="submit">Save</button>
-      </fieldset>
-    </form>
+    <template is="dom-if" if="[[addUserSelected]]">
```
Comment (Jropp) [inline src/components/create-user.html]: This seems like it should be opposite. Like when you clicked the button that means you've selected addUser.

File: src/components/create-user.html [hunk 2]
```diff
@@ -44,31 +42,90 @@
...
+          <td><label for="firstName">First name :</label></td>
+          <td><input type="text" name="firstName" id="firstName" value="{{user.firstName}}" on-input="updateFirstName"></td>
+        </tr>
+        <tr>
+          <td><label for="lastName">Last name :</label></td>
+          <td><input type="text" name="lastName" id="lastName" value="{{user.lastName}}" on-input="updateLastName"></td>
+        </tr>
+        <tr>
+          <td><label for="phone">Phone :</label></td>
+          <td><input type="text" name="phone" id="phone" value="{{user.phoneNumber}}" on-input="updatePhoneNumber"></td>
```
Comment (Jropp) [inline src/components/create-user.html]: Do you need two way data binding here? (See the comment below)

File: src/components/create-user.html [hunk 3]
```diff
@@ -44,31 +42,90 @@
...
+
+      createNewUserProfile(event) {
+        this.addUserSelected = !this.addUserSelected;
+      }
+
+      changeButtonText(addUserSelected) {
+        return addUserSelected ? `Create User` : `Save`;
+      }
+
+      updateFirstName(event) {
```
Comment (Jropp) [inline src/components/create-user.html:114]: If you are using the two way data binding above, are these functions even necessary?

**Review by Jropp — CHANGES_REQUESTED** [review]
I didn't have time to get as deep as I wanted into some parts, I'll look at it a bit more later but I thought this at least started things off.

File: src/components/create-user.html (hunk 2 above)
Comment (nathanjophillips) [reply src/components/create-user.html]: I think you're right I might just need one, I'll see if it still works

File: src/components/create-user.html (hunk 1 above)
Comment (nathanjophillips) [reply src/components/create-user.html]: yeah I can change that, it makes more sense that way for sure

**Review by Jropp — APPROVED** [review]
Looks great! Works as expected.

**Pull request merged**